### ISSUE TITLE
Looking up a deactivated widget's ancestor is unsafe

### ISSUE THREAD

**Issue by z234009184** [chudongvip/awesome_video_player#11]
════════ Exception caught by scheduler library ═════════════════════════════════════════════════════
The following assertion was thrown during a scheduler callback:
Looking up a deactivated widget's ancestor is unsafe.

At this point the state of the widget's element tree is no longer stable.

To safely refer to a widget's ancestor in its dispose() method, save a reference to the ancestor by calling dependOnInheritedWidgetOfExactType() in the widget's didChangeDependencies() method.

When the exception was thrown, this was the stack: 
#0      Element._debugCheckStateIsActiveForAncestorLookup.<anonymous closure> (package:flutter/src/widgets/framework.dart:3508:9)
#1      Element._debugCheckStateIsActiveForAncestorLookup (package:flutter/src/widgets/framework.dart:3522:6)
#2      Element.dependOnInheritedWidgetOfExactType (package:flutter/src/widgets/framework.dart:3564:12)
#3      MediaQuery.of (package:flutter/src/widgets/media_query.dart:793:38)
#4      _AwsomeVideoPlayerState.initState.<anonymous closure>.<anonymous closure> (package:awsome_video_player/src/video.dart:113:38)
...
════════════════════════════════════════════════════════════════════════════════════════════════════

**Comment by z234009184** [comment]
作者您好,非常感谢您开源这么强大的库!!
我在使用的过程中遇到了这个错误, 可以帮我看下什么原因么?
感谢

**Comment by chudongvip** [comment]
> 者您好,非常感谢您开源这么强大的库!!
> 我在使用的过程中遇到了这个错误, 可以帮我看下什么原因么?
> 感谢

收到，请问你用的是模拟器还是真机，系统版本以及型号提供一下

**Comment by z234009184** [comment]
> > 者您好,非常感谢您开源这么强大的库!!
> > 我在使用的过程中遇到了这个错误, 可以帮我看下什么原因么?
> > 感谢
> 
> 收到，请问你用的是模拟器还是真机，系统版本以及型号提供一下

您好, 我使用的是 iPhone 7 真机,  iOS 13.3.1  
发生在 onpop 回调中调用 Navigator.pop(context);

**Comment by z234009184** [comment]
经过测试, 我发现好像是在 Navigator.pop(context) 之后, 播放器内部的MediaQuery.of 不断的使用 context 而引发, 不知道是不是这样

**Comment by chudongvip** [comment]
> 经过测试, 我发现好像是在 Navigator.pop(context) 之后, 播放器内部的MediaQuery.of 不断的使用 context 而引发, 不知道是不是这样

好的，安排。ps: 你是单页面还是多页面，播放器是在首页还是子页面调用 Navigator.pop

**Comment by z234009184** [comment]
> 好的，安排。ps: 你是单页面还是多页面，播放器是在首页还是子页面调用 Navigator.pop

您好, 播放器是在子页面进行的 Navigator.pop(context)

**Comment by seekcx** [comment]
一样的问题，不知道是不是我在一个list里使用了多个导致的。

**Comment by chudongvip** [comment]
> 一样的问题，不知道是不是我在一个list里使用了多个导致的。

等我发布新的版本，你再试试，今天晚上发一版

**Comment by niicolasalves** [comment]
Hello, I have the same problem. In my case, I am using a PageView and when leaving the page with the player and returning it is in a loop repeating this error.


```
════════ Exception caught by scheduler library ═════════════════════════════════
Looking up a deactivated widget's ancestor is unsafe.
════════════════════════════════════════════════════════════════════════════════
```

Running on a real device on Android 9.

**Comment by chudongvip** [comment]
> Hello, I have the same problem. In my case, I am using a PageView and when leaving the page with the player and returning it is in a loop repeating this error.
> 
> ```
> ════════ Exception caught by scheduler library ═════════════════════════════════
> Looking up a deactivated widget's ancestor is unsafe.
> ════════════════════════════════════════════════════════════════════════════════
> ```
> 
> Running on a real device on Android 9.

Thanks for your feedback. This is a know BUG that i have fixed and i'll release a version today.

**Comment by chudongvip** [comment]
@niicolasalves @z234009184  I'll fixed this bug with the version 1.0.7
